core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -646,7 +645,17 @@ public ParseField getParseField() {
             return parseField;
         }
 
-        public ScriptMode getDefaultScriptMode() {
+        public ScriptMode getDefaultScriptMode(@Nullable String language) {
+            if (language != null) {
```

Comment:
I pushed a commit that changes the script engine registration to take a value for whether the script should be enabled or disabled by default so they don't have to be special-cased in the script service.